modules/ibc-hooks/wasm_hook.go

```diff
@@ -181,7 +177,6 @@ func ValidateAndParseMemo(memo string, receiver string) (isWasmRouted bool, cont
 			fmt.Errorf(types.ErrBadMetadataFormatMsg, memo, "wasm metadata is not a valid JSON map object")
 	}
 
-	// Get the contract
```

Comment:
this comment doesn't bring anything. The implementation and the error message is clear and short, so it's better to read the code than the comment.
[Rule 1: Comments should not duplicate the code](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments)

Reply:
same comment applies to few changes below